File: javascript/2021/day1.mjs
import { readFileSync } from 'node:fs'

const data = readFileSync('./2021/resources/day1.txt', { encoding: 'utf8' }).split('\n')
    .map(Number);

export function part1() {
    let counter = 0;

    for (let i = 0, l = data.length - 1; i < l; ++i) {
        if (data[i + 1] > data[i]) ++counter;
    }

    return counter;
}

export function part2() {
    let counter = 0;

    for (let i = 0, l = data.length - 2; i < l; ++i) {
        if (
            (data[i + 1] + data[i + 2] + data[i + 3]) > (data[i] + data[i + 1] + data[i + 2])
        ) ++counter;
    }

    return counter
}

File: rust-src/src/year_2021/day1.rs
use std::fs;

fn load_file() -> Vec<u16> {
    fs::read_to_string("src/year_2021/resources/day1.txt")
        .unwrap()
        .lines()
        .map(|line| line.parse::<u16>().unwrap())
        .collect()
}

pub fn part1() -> u16 {
    let numbers = load_file();
    let mut counter: u16 = 0;

    for (i, n) in numbers[0..numbers.len() - 1].iter().enumerate() {
        if numbers[i + 1] > *n {
            counter = counter + 1;
        }
    }

    counter
}

//pub fn part2() {}
